### PR TITLE
[3.4] Bug fix 3.4/add shard id to replication client identifier

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/9366
There also are related changes in https://github.com/arangodb/arangodb/pull/9473

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [X] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [X] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [ ] Added **Regression Tests** (Only for bug-fixes) 
- [ ] Added new C++ **Unit Tests** (Either GoogleTest or Catch-Test)
- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
- [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

### Documentation

- [x] Added a *Changelog Entry* 
- [x] Added a new section in the *http API*

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5187/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5200/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5220/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5224/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5250/

**Comment by goedderz** [comment]
Last PR run had one test failure in enterprise/rocksdb/cluster:
```
* Test "resilience_failover"
    [FAILED]  tests/js/server/resilience/failover/resilience-synchronous-repl-cluster.js

      "testLargeTransactionsSplitting" failed: ArangoError 1457: timeout in cluster operation - ArangoError: timeout in cluster operation
    at ArangoDatabase._drop (/work/ArangoDB/js/server/modules/@arangodb/arango-database.js:182:23)
    at Object.tearDown (tests/js/server/resilience/failover/resilience-synchronous-repl-cluster.js:353:10)
```
Looks unrelated, but I'm trying to reproduce it locally and restarted the linux PR matrix: http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6914/

**Comment by goedderz** [comment]
I could not reproduce it locally, Jenkins is still running. Even if that's green, I'd like a second opinion whether this could be related.

File: arangod/VocBase/vocbase.h [hunk 1]
```diff
@@ -166,8 +168,10 @@ struct TRI_vocbase_t {
 
   std::unique_ptr<arangodb::DatabaseReplicationApplier> _replicationApplier;
 
-  arangodb::basics::ReadWriteLock _replicationClientsLock;
-  std::unordered_map<TRI_server_id_t, std::tuple<double, double, TRI_voc_tick_t>> _replicationClients;
+  // arangodb::basics::ReadWriteLock _replicationClientsLock;
```
Comment (jsteemann) [inline arangod/VocBase/vocbase.h]: let's remove the old definitions here

**Comment by jsteemann** [comment]
Let's put this into 3.4.9

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6072/

**Comment by goedderz** [comment]
Jenkins is green. http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/6118/

**Pull request merged**